### ISSUE TITLE
Adding some files fail with "Only FlateDecode filter are supported"

### ISSUE THREAD

**Issue by kometbomb** [rkusa/pdfjs#268]
I have a PDF file that makes the lib throw "Only FlateDecode filter are supported" when adding the file. The cause seems to be that the code arrives at [lib/util.js:40](https://github.com/rkusa/pdfjs/blob/ba739922ec01c3b67ba9f969bef88fbf1509cc5b/lib/util.js#L40) with the filter being an array (that contains just the one FlateDecode filter). 

```
/Filter [FlateDecode]
```

A fix that seems to correct this issue is to change the above line to:
```
  let filter =   let filter = obj.properties.get("Filter")[0] || obj.properties.get("Filter")
```

I have no idea why this happens and if this is the correct way to fix it, though. And I have no idea if the PDF in question is valid. The PDF is from a client so I can't share it.

**Comment by rkusa** [comment]
I've checked the PDF specification, the field can indeed be an array, so checking whether it is an array, making sure it has only one single entry and ensuring that that entry equals `FlateDecode` would be a correct fix.

**Comment by kometbomb** [comment]
Ah ok, I was worrying the files itself are somehow weird. It seems very straight-forward to fix but let me know if I can help in any way.

**Comment by rkusa** [comment]
Should be fixed in `v2.4.7`, please let me know if not. Thanks again for report

**Comment by kometbomb** [comment]
Thank you!

ti 9. marrask. 2021 klo 11.06 Markus Ast ***@***.***)
kirjoitti:
